src/bunit.core/Extensions/WaitForHelpers/WaitForHelper.cs

```diff
@@ -180,11 +207,24 @@ private void OnAfterRender(object? sender, EventArgs args)
 						renderedFragment.RenderCount,
 						renderer.RenderCount,
 						capturedException));
-				Dispose();
+			}
+
+			if (MinimumRenderCountPassed())
+			{
+				checkPassedCompletionSource.TrySetException(
+					new WaitForFailedException(
+						MaxRenderCountErrorMessage ?? string.Empty,
+						checkCount,
+						renderedFragment.RenderCount,
+						renderer.RenderCount,
+						capturedException));
 			}
 		}
 	}
 
+	private bool MinimumRenderCountPassed()
```

Comment:
Why `MinimumRenderCountPassed`? That name is confusing to me.

```suggestion
	private bool MaximumRenderCountReached()
```

Reply:
yeah, thats a leftover from a different version.